\documentclass{beamer}
\usepackage{graphicx}
\usepackage{float}
\usepackage{amsmath}
\usepackage{caption}
\usepackage{subcaption}
\usepackage{setspace}
\usepackage[english,ngerman]{babel}
\usepackage[T1]{fontenc}
\usepackage{tikz}
\usepackage{epigraph}
\captionsetup[figure]{labelformat=empty}
\selectlanguage{english}
\graphicspath{{../Figures/}}
\usetheme{Dresden}
\usecolortheme{spruce}
\useoutertheme{infolines}
\useinnertheme{rectangles}
\setbeamercovered{transparent} 
\colorlet{mdtRed}{red!50!black}
\beamertemplatenavigationsymbolsempty 

\usepackage{tikz}
\usetikzlibrary{shapes.geometric, arrows}
\tikzstyle{startstop} = [rectangle, rounded corners, minimum width=3cm, minimum height=1cm,text centered, draw=black, fill=red!30]
\tikzstyle{io} = [trapezium, trapezium left angle=70, trapezium right angle=110, minimum width=3cm, minimum height=1cm, text centered, draw=black, fill=blue!30]
\tikzstyle{process} = [rectangle, minimum width=3cm, minimum height=1cm, text centered, draw=black, fill=orange!30]
\tikzstyle{decision} = [diamond, minimum width=3cm, minimum height=1cm, text centered, draw=black, fill=green!30]
\tikzstyle{arrow} = [thick,->,>=stealth]




 \title[Quantitative genetics] %optional
{\color{mdtRed}{Quantitative genetics from genome assemblies to neural network aided omics-based prediction of complex traits}}
 
 
\author{Jan Freudenthal}
 
\institute[CCTB] % (optional)
{
  % 
  CCTB \\
  Evolutionary genomics \\
  Julius-Maximilians-Universität Würzburg
}
 
\date{31. Jan 2020}
 
%\logo{\includegraphics[height=1.5cm]{lion-logo.png}}

\begin{document}
\frame{\titlepage}

\begin{frame}
  \frametitle{Quantitative genetics}
  Quantitative genetics aims to explain the heritable parts of traits that follow certain statistical distributions.
\end{frame}


\begin{frame}
    \frametitle{Quantitative genetics}
    \includegraphics[width=.98\textwidth, height=.8\textheight]{../Figures/gxe.png}
\end{frame}

\begin{frame}
    \frametitle{Complex trait}
    \includegraphics[width=.98\textwidth, height=.8\textheight]{../Figures/complex_trait.png}
\end{frame}

\begin{frame}
    \frametitle{Quantitative genetics}
    \includegraphics[width=.98\textwidth, height=.8\textheight]{../Figures/athal.png}
\end{frame}



\begin{frame}
  \begin{huge}
    \frametitle{Decomposition of phenotypic variance} \vspace{-2ex}
    \begin{itemize}[<+->]
    \item[] \[\sigma_P = \sigma_G + \sigma_E + \sigma_{GxE}\]
    \item[] \[\sigma_G = \sigma_A + \sigma_D + \sigma_I\]
    \item[] \[\sigma_I = \sigma_{AA} + \sigma_{AD} + \sigma_{DD}\]
    \item[] \[h^2 = \frac{\sigma_A}{\sigma_P} \]
    \end{itemize}
  \end{huge}
\end{frame}
 



\begin{frame}
  \frametitle{Workflow in quantitative Genetics}
  
\begin{figure}[H]
  \begin{center}
    \begin{tikzpicture}[node distance=2cm, scale=0.8, transform shape]
      \node (start0) [startstop] {Selection of suitable candidates};
      \node (start) [startstop,below of=start0] {DNA extraction};
      \draw [arrow] (start0) -- (start);
      \node (seq) [process, below of=start, xshift=-3cm] {Sequencing} ;
      \draw [arrow] (start) -- (seq);
      \node (SNP) [process, below of=start, xshift=3cm, yshift=-2cm] {SNP array} ;
      \draw [arrow] (start) -- (SNP);
      \node (ga) [process, below of=seq] {Genome assembly} ;
      \draw [arrow] (seq) -- (ga);
      \node (snpca) [process, below of=ga] {Alignment \& SNP calling} ;
      \draw [arrow] (ga) -- (snpca);
      \node (imp) [io, below of=snpca, xshift=3cm] {Imputation of missing values};
      \draw [arrow] (snpca) -- (imp) ;
      \draw [arrow] (SNP) -- (imp) ;
      \node (LD) [io, below of=imp] {LD pruning} ;
      \draw [arrow] (imp) -- (LD) ;
      \node (MAF) [io, below of=LD] {MAF filtering} ;
      \draw [arrow] (LD) -- (MAF) ;
      \node (bm) [startstop, below of=MAF] {Numeric marker matrix} ;
      \draw [arrow] (MAF) -- (bm) ;
    \end{tikzpicture}
    \caption[Schematic process of genotyping for quantitative genetics]{Schematic process of genotyping for quantitative genetics analyses with its crucial steps} \label{fig:quan_flow1}
  \end{center}     
\end{figure}
\end{frame}

\begin{frame}
\vspace{-22em}
\begin{figure}[H]
  \begin{center}
    \begin{tikzpicture}[node distance=2cm, scale=0.8, transform shape]
      \node (start0) [startstop] {Selection of suitable candidates};
      \node (start) [startstop,below of=start0] {DNA extraction};
      \draw [arrow] (start0) -- (start);
      \node (seq) [process, below of=start, xshift=-3cm] {Sequencing} ;
      \draw [arrow] (start) -- (seq);
      \node (SNP) [process, below of=start, xshift=3cm, yshift=-2cm] {SNP array} ;
      \draw [arrow] (start) -- (SNP);
      \node (ga) [process, below of=seq] {Genome assembly} ;
      \draw [arrow] (seq) -- (ga);
      \node (snpca) [process, below of=ga] {Alignment \& SNP calling} ;
      \draw [arrow] (ga) -- (snpca);
      \node (imp) [io, below of=snpca, xshift=3cm] {Imputation of missing values};
      \draw [arrow] (snpca) -- (imp) ;
      \draw [arrow] (SNP) -- (imp) ;
      \node (LD) [io, below of=imp] {LD pruning} ;
      \draw [arrow] (imp) -- (LD) ;
      \node (MAF) [io, below of=LD] {MAF filtering} ;
      \draw [arrow] (LD) -- (MAF) ;
      \node (bm) [startstop, below of=MAF] {Numeric marker matrix} ;
      \draw [arrow] (MAF) -- (bm) ;
    \end{tikzpicture}
    \caption[Schematic process of genotyping for quantitative genetics]{Schematic process of genotyping for quantitative genetics analyses with its crucial steps} \label{fig:quan_flow1}
  \end{center}     
\end{figure}
\end{frame}


\begin{frame}
  \frametitle{Numeric marker matricies}
   \includegraphics[height=.8\textheight,width=.98\textwidth]{../Figures/chr1_hap}
\end{frame}

\begin{frame}
  Schematic representation of a genotype matrix as used in genetic analysis like GWAS and genomic prediction \\
\begin{table}[H]
 \centering
 \begin{tabular}{l|cccc} 
   \hline
        & M-1 & M-2 & M-3 & M-4 \\
   \hline
   Acc1 & 0   & 1   & 1   & 0   \\
   Acc2 & 1   & 0   & 1   & 0   \\
   Acc3 & 0   & 1   & 0   & 1   \\
   Acc4 & 1   & 0   & 0   & 1   \\
   \hline
 \end{tabular}
\end{table}
\end{frame}


\begin{frame}
  \frametitle{Methods in quantitative genetics}
  \includegraphics[height=.8\textheight,width=.9\textwidth]{bla.pdf}
\end{frame}

\begin{frame}
  \frametitle{Objectives}
  \begin{enumerate} [<+->]
  \item Improve GWAS methodology 
  \item Apply non-parametric statistical methods to genomic selection
  \end{enumerate}
\end{frame}

\begin{frame}
  \frametitle{GWAS}
  \begin{itemize}[<+->]
  \item GWAS is the main method used to link traits/phenotypes to genetic polymorphisms
  \item GWAS utilizes mixed-model linear equations to account for structured populations
  \item Significant testing is commonly done with bonferroni thresholds 
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Problems with GWAS and bonferroni correction}
  \begin{itemize}[<+->]
  \item With increasing phenotypes and markers the computational time increases exponentially
  \item Bonferroni assumes independent testing
  \item Due to LD markers are not independent from each other
  \item Permutation based thresholds are better suited to account for LD and structured population
  \item Permutations have to be repeated 100 times with shuffled phenotypes
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{GWAS Flow}
  \begin{itemize}[<+->]
  \item GWAS consists of a series of matrix operations that can be highly parallelized
  \item GWAS Flow uses the TensorFlow's Python API 
  \item The calculations can be run on both GPU and CPU 
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Performance of GWAS-Flow }
  \includegraphics[height=.8\textheight,width=.9\textwidth]{final_plot2.pdf}
\end{frame}


\begin{frame}
  \frametitle{Genomic selection}
  \includegraphics[height=.8\textheight,width=.9\textwidth]{gs}
\end{frame}

\begin{frame}
  \frametitle{Advantages of genomic selection}
  \begin{enumerate}[<+->]
  \item Selection from larger populations
  \item Stricter selection intensity 
  \item Acceleration of the breeding cycle
  \item Reduction in phenotyping costs
  \end{enumerate}
\end{frame}


\begin{frame}
  \frametitle{Prediction methods in genomic selection}
  \begin{enumerate}[<+->]
  \item GBLUP - the gold-standard in genomic selection
  \item Bayesian methods
  \item Bayesian methods differ in the a prior assumptions on marker effect sizes
  \item RHKS, random forest, support vector machines etc.
  \end{enumerate}    
\end{frame}
 
\begin{frame}
   \frametitle{Neural networks in genomic prediction}
   \begin{columns}
     \begin{column}{0.5\textwidth}
       \begin{itemize}
       \item With the advent of high performing GPUs neural networks become popular in all branches
         \item In the scope of the present study the usability of neural networks for genomic selection was assessed 
       \end{itemize}
     \end{column}
     \begin{column}{0.5\textwidth}
        \includegraphics[height=.6\textheight,width=.9\textwidth]{neuralnet}
     \end{column}
   \end{columns}
\end{frame}
 

\begin{frame}
  \frametitle{Datasets}
  \begin{itemize}[<+->]
  \item \textit{A. thaliana} data from the 1001 genome project with 10 mio markers and 164 different phenotypes
  \item Doubled-haploid maize populations derived from two maize landraces
  \end{itemize}
\end{frame}


\begin{frame}
  \frametitle{Genomic prediction \textit{A. thaliana}}
  \includegraphics[height=.8\textheight,width=.9\textwidth]{ann_vs_gblup}
\end{frame}

\begin{frame}
  \frametitle{Genomic selection in maize landraces}
  \includegraphics[height=.8\textheight,width=.9\textwidth]{gp_kemater}
\end{frame}


\begin{frame}
  \frametitle{Number of markers versus prediction accuracy}
  \includegraphics[height=.8\textheight,width=.9\textwidth]{marker_vs_acc}
\end{frame}

\begin{frame}
\frametitle{Number of phenotypes versus prediction accuracy}
  \includegraphics[height=.8\textheight,width=.9\textwidth]{phenos_acc}
\end{frame}

\begin{frame}
  \frametitle{Comparison of bayesian methods}
  \includegraphics[height=.8\textheight,width=.9\textwidth]{pred_acc_bayes}
\end{frame}


\begin{frame}
  \frametitle{Conclusions}
  \begin{itemize} [<+->]
  \item Neural networks are well-suited for genomic selection
  \item All methods fail to outperform all the other methods
  \item The limiting factors are the number of accessions and the
    effective population size
  \item Mostly additive effects are captured by the prediction equations
  \end{itemize}
\end{frame}


\begin{frame}
Thanks for your attention!
\end{frame}


\end{document}
%%% Local Variables:
%%% mode: latex
%%% TeX-master: t
%%% End:
