% Chapter 1

\chapter{Benchmarking of Chloroplast Genome Assembly tools } % Main chapter title

\label{Chapter1} % For referencing the chapter elsewhere, use \ref{Chapter1}
This chapter orientates on \cite{freudenthal2019landscape} only the chapters from the publication which the author majorly contributed to are included. 

%----------------------------------------------------------------------------------------

% Define some commands to keep the formatting separated from the content 
\newcommand{\keyword}[1]{\textbf{#1}}
\newcommand{\tabhead}[1]{\textbf{#1}}
\newcommand{\code}[1]{\texttt{#1}}
\newcommand{\file}[1]{\texttt{\bfseries#1}}
\newcommand{\option}[1]{\texttt{\itshape#1}}

%----------------------------------------------------------------------------------------

\section{Introduction}

Circular DNA of a size between 120 kBP to 160 kBP \cite{palmer_1985}.  First choloroplast sequenced as early
as 1986 \textit{Marchantia polymorpha} and \textit{Nico} \cite{ohyama_chloroplast_1986}; \cite{shinozaki_complete_1986}.
Review  genome structure \cite{green_chloroplast_2011}; \cite{wicke_evolution_2011}.
Chloroplast genomes widely used in evolutionary studies \cite{martin_evolutionary_2002}; \cite{xiao-ming_inferring_2017}.
Chloroplast genomes are small through endosymbiotic gene transfer \cite{martin_evolutionary_2002}; \cite{deiner_environmental_2017}. Up to 14 \% of the the core genome of \textit{Arabidopsis thaliana} is made up of genes previously from the chloroplast (fancy citation), while 100-120 genes remain on the chloroplast \cite{wicke_evolution_2011}.


%----------------------------------------------------------------------------------------
\section{Material and Methods}
\subsection{Methods}
\subsection{Tools}
\subsection{Evaluation}
\subsubsection{Quantitative}
\begin{equation}
  score = \frac{1}{4} \cdot \left( cov_{ref} +  cov_{qry} + min\left\{ \frac{cov_{qry}}{cov_{ref}}, \frac{cov_{ref}}{cov_{qry}}\right\} + \frac{1}{n_{contigs} }\right) \cdot 100 
\label{eqn:score_ass}
\end{equation}

\subsubsection{Qualitative}
\subsubsection{Consisteny}


\subsection{Data}
\subsubsection{Simulated}
\subsubsection{Real data set}
\subsubsection{Novel data set}


\section{Results}
\subsection{Qualitative}

\subsection{Quantitatve}
\subsubsection{Simulated data}
\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/sim_tiles}
\decoRule
\caption[Score of assemblies of simulated data sets]{Results of assemblies exectuted with simulated data sets.}
\label{fig:sim_tiles}
\end{figure}



\begin{table}[h!]
\caption{\textbf{Scores of assemblies of simulated data}}
\label{tab:scores_simulated}
\centering
\begin{tabular}{rlrrrrrrr}
  \hline
 & data set & CAP & CE & Fast-Plast & GetOrganelle & IOGA & NOVOPlasty & org.ASM \\ 
  \hline
 1 & sim\_150bp.0-1 & 79.10 & 100.00 & 99.48 & 100.00 &  & 91.52 & 100.00 \\ 
  2 & sim\_150bp.0-1.2M & 79.10 & 100.00 & 99.72 & 100.00 & 79.10 & 91.52 & 91.50 \\ 
  3 & sim\_150bp.1-10 &  & 56.44 & 100.00 & 76.98 &  & 91.52 & 78.00 \\ 
  4 & sim\_150bp.1-10.2M &  &  & 99.97 & 100.00 &  & 91.52 & 82.72 \\ 
  5 & sim\_150bp.1-100 & 75.72 & 100.00 & 99.48 & 100.00 & 66.09 & 91.52 & 91.50 \\ 
  6 & sim\_150bp.1-100.2M &  & 100.00 & 99.47 & 100.00 &  & 100.00 & 100.00 \\ 
  7 & sim\_150bp.1-1000 & 79.10 &  & 99.72 & 100.00 &  & 91.52 & 100.00 \\ 
  8 & sim\_150bp.1-1000.2M & 79.10 & 100.00 & 99.72 & 100.00 &  & 91.52 & 100.00 \\ 
  9 & sim\_250bp.0-1 & 79.10 & 100.00 & 93.82 & 100.00 &  & 91.52 & 91.50 \\ 
  10 & sim\_250bp.0-1.2M & 79.10 & 100.00 & 93.83 & 100.00 &  & 91.52 & 91.50 \\ 
  11 & sim\_250bp.1-10 &  & 54.98 & 68.45 & 78.89 & 52.71 & 91.52 & 40.20 \\ 
  12 & sim\_250bp.1-10.2M &  &  & 93.00 & 100.00 & 52.67 & 87.40 & 40.20 \\ 
  13 & sim\_250bp.1-100 & 72.81 & 100.00 & 93.82 & 100.00 &  & 87.40 & 100.00 \\ 
  14 & sim\_250bp.1-100.2M &  & 100.00 & 93.83 & 100.00 &  & 87.40 & 100.00 \\ 
  15 & sim\_250bp.1-1000 & 79.10 & 21.30 & 93.83 & 100.00 & 76.96 & 91.52 & 91.50 \\ 
  16 & sim\_250bp.1-1000.2M & 79.10 & 100.00 & 93.83 & 100.00 & 67.55 & 87.40 & 100.00 \\
   \hline
\end{tabular}
\end{table}







\subsubsection{Real data sets}
\begin{table}[h!]
\caption{\textbf{Mean scores of chloroplast genome assemblers}}
\label{tab:scores_real}
\centering
\begin{tabular}{rlrrrr}
  \hline
 & assembler & Median & IQR & N\_perfect & N\_tot \\ 
  \hline
1 & CAP & 45.25 & 50.19 &   0 & 369 \\ 
  2 & CE & 56.55 & 71.50 &  14 & 369 \\ 
  3 & Fast-Plast & 92.80 & 23.59 & 113 & 369 \\ 
  4 & GetOrganelle & 99.83 & 20.94 & 210 & 360 \\ 
  5 & IOGA & 71.10 & 11.21 &  0 & 338 \\ 
  6 & NOVOPlasty & 75.95 & 48.69 &  58 & 369 \\ 
  7 & org.ASM & 67.35 & 91.69 &  46 & 348 \\ 
   \hline
\end{tabular}
\end{table}





\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/swarm}
\decoRule
\caption[Scores of assemblies from real data sets]{Box and swarm plots depict the resluts from the scoring shown in \ref{eqn:score_ass}}
\label{fig:swarm}
\end{figure}




\subsubsection{Consistency}

\subsubsection{Real data sets}
\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/repro}
\decoRule
\caption[Comparison between two uns with the same assembler for consistency testing ]{Swarm plots depict the resluts from the scoring shown in \ref{eqn:score_ass} for two independet runs for each assembler on each of the datasets}
\label{fig:consisplot}
\end{figure}


\subsubsection{Novel}


\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/upset_novel}
\decoRule
\caption[Upset plot comparing the success rates for novel data sets]{nls}
\label{fig:upset_novel}
\end{figure}

\section{Disucssion}

\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/upset}
\decoRule
\caption[Upset plot comparing the success rates of of all assemblers]{Upset plot showing the intersections of sucess rates between assemblers. A successful assembly was defined with a score > 99 according to equation \ref{eqn:score_ass}}
\label{fig:upset}
\end{figure}


