% Chapter 1
\newcommand{\formatprogramnames}[1]{\texttt{#1}}
\newcommand{\ce}{\formatprogramnames{chloroExtractor}}
\newcommand{\oa}{\formatprogramnames{ORG.Asm}}
\newcommand{\fp}{\formatprogramnames{Fast-Plast}}
\newcommand{\ioga}{\formatprogramnames{IOGA}}
\newcommand{\np}{\formatprogramnames{NOVOPlasty}}
\newcommand{\go}{\formatprogramnames{GetOrganelle}}
\newcommand{\cassp}{\formatprogramnames{Chloroplast assembly protocol}}



\chapter{Benchmarking of Chloroplast Genome Assembly tools } % Main chapter title

\label{Chapter1} % For referencing the chapter elsewhere, use \ref{Chapter1}
This chapter orientates on \cite{freudenthal2019landscape} only the chapters from the publication which the author majorly contributed to are included. 

%----------------------------------------------------------------------------------------

% Define some commands to keep the formatting separated from the content 
\newcommand{\keyword}[1]{\textbf{#1}}
\newcommand{\tabhead}[1]{\textbf{#1}}
\newcommand{\code}[1]{\texttt{#1}}
\newcommand{\file}[1]{\texttt{\bfseries#1}}
\newcommand{\option}[1]{\texttt{\itshape#1}}

%----------------------------------------------------------------------------------------

\section{Introduction}
\subsection{Motivation}

Circular DNA of a size between 120 kBP to 160 kBP \cite{palmer_1985}.  First chloroplast sequenced as early as
1986 \textit{Marchantia polymorpha} and \textit{Nico} \cite{ohyama_chloroplast_1986};
\cite{shinozaki_complete_1986}.  Review genome structure \cite{green_chloroplast_2011};
\cite{wicke_evolution_2011}.  Chloroplast genomes widely used in evolutionary studies
\cite{martin_evolutionary_2002}; \cite{xiao-ming_inferring_2017}.  Chloroplast genomes are small through
endosymbiotic gene transfer \cite{martin_evolutionary_2002}; \cite{deiner_environmental_2017}. Up to 14 \% of
the the core genome of \textit{Arabidopsis thaliana} is made up of genes previously from the chloroplast
(fancy citation), while 100-120 genes remain on the chloroplast \cite{wicke_evolution_2011}.  However
chloroplast between plant species show large variety e.g. parasitic plants.  Chloroplast genomes are much
smaller than core genomes e.g \textit{A. thaliana} ca. 125 MBP. Highly conserved high gene content, therefore
changes are more likely to be functional.  Single chloroplast contain up to hundreds of copies of its genome
\cite{kumar_2014}; \cite{bendich_1987}, and photosynthetic activate cells contain multiple chloroplasts,
therefore the copy number of chloroplast genomes is much higher than number of core genomes per cell.
Structurally chloroplast genomes consist of two inverted repeats (IR) - $IR_A$ and $IR_B$ - ranging from 10
kbp to 76 kB the divide the chloroplast genome into two distinct regions the large single copy (LSC) and the
small single copy (SSC) as shown in \ref{fig:cpast_genome} \cite{palmer_1985}. Taking into account that the
majority of assembly tools has been designed to assemble linear core genomes, the structure of chloroplast
genomes is a major obstacle when wanting to assemble those genomes with modern short read technologies,
especially solving and aligning the IR \cite{Wang2018}. 

\begin{figure}[H]
\centering
\includegraphics[height=.55\textheight, width=.95\textwidth]{Figures/cpast}
\decoRule
\caption[Structure of a chloroplast genome]{Structure of the chloroplast genome of \textit{A. thaliana} with SSC-Region and LSC-Region. Length of the genome and its parts in kbp. Graphic from \cite{olejniczak2016chloroplasts}}
\label{fig:cpast_genome}
\end{figure}

Chloroplast genomes present biologist with many obstacles of which heteroplasmy stands out and immensely
complicated genome assemblies and downstream analyses \cite{corriveau_1988}; \cite{Chat2002}.  Heteroplasmy
describes the phenomenon of co-existence of multiple versions of the chloroplast genomes in a single organism
and single cells of that respective organism. The underlying evolutionary mechanisms behind heteroplasmy are no
fully elucidated and eventually existing fitness advantages fueling heteroplasmy cannot be explained
satisfactory by evolutionary methods \cite{Scar2016}.
There is a variety of databases containing short read data for species without an chloroplast genome available. For example NCBI's the short read archive (SRA) \cite{SRA2010}. Because most plant DNA extraction protocols use green leaf tissue as basis, they usually contain a large amount of plastid DNA. Having  larger number of assembled and annotated chloroplast genomes publicly available would be beneficial for evolutionary studies and could be a useful addition to bar-coding and super-barcoding \cite{coissac_barcodes_2016} and other biotechnological applications \cite{daniell_chloroplast_2016}. To achieve this there is a variety of tools available. The study presented in this chapter, assesses the availability, usability and overall performance of 7 of those tools and ultimately makes use of the newly gained insights to attempt \textit{de novo} assemble more than 100 chloroplast




\subsection{Extraction of chloroplast reads from whole genome data and general assembly workflow}
There is a variety of strategies to assemble chloroplast genomes from raw sequencing data \cite{twyford_strategies_2017}. In general the process involves three steps: (i) extraction of plastid reads from the WGS data, (ii) assembly of the plastid genome, (iii) solving the circular structure of the chromosome with the IRs. There are two distinct ways to tackle step (i): The first one is to map all the reads to a reference chloroplast \cite{Vinga2012}, which works reasonably well if there is one available for the same, or at least a closely related species. The second is to make use of the much larger coverage of chloroplast DNA compared to core DNA, with a k-mer analysis \cite{Chan2013}, this is for example done by \ce \cite{j_ankenbrand_chloroextractor:_2018}. The third way to accomplish this is to combine as done by \np  \cite{dierckxsens_novoplasty:_2017}.
Figure \ref{fig:cpast_worklow} shows the general workflow of chloroplast assembly too. 


\begin{figure}[H]
\centering
\includegraphics[height=.65\textheight, width=.95\textwidth]{Figures/CE_workflow}
\decoRule
\caption[Chloroplast genom assembly workflow]{Graphic from \cite{j_ankenbrand_chloroextractor:_2018} }
\label{fig:cpast_workflow}
\end{figure}


\subsubsection{Purpose and scope of benchmarking the landscape of chloroplast assembly tools}

The purpose of this study is to provide insights into the landscape of chloroplast assembly tools, to recommend best practices for organelle genome assemblies and to provide \textit{de novo} assemblies for many species and family's without a reference chloroplast available so far.
To be included into this study the software, including the source code must be publicly available and published under the terms of a liberal free open source software e.g. GPL or MIT-license. The study was also limited to paired-end Illumina data sets as their sole input source, because they are abundantly available for this benchmark.  The seven tools that me the requirements and were therefore included in this study were: \ce,  \oa,  \fp,  \ioga,  \np,  \go and  \cassp. As later thoroughly described in section \ref{results:ca}, there are huge differences between those tools and some will not be recommend to be used in scientific applications, while others outstand for most but not all cases.


%----------------------------------------------------------------------------------------
\section{Material and Methods}
\subsection{Methods}
\subsection{Tools}
\subsection{Evaluation}
\subsubsection{Quantitative}
\begin{equation}
  score = \frac{1}{4} \cdot \left( cov_{ref} +  cov_{qry} + min\left\{ \frac{cov_{qry}}{cov_{ref}}, \frac{cov_{ref}}{cov_{qry}}\right\} + \frac{1}{n_{contiguous} }\right) \cdot 100 
\label{eqn:score_ass}
\end{equation}

\subsubsection{Qualitative}
\subsubsection{Consistency}


\subsection{Data}
\subsubsection{Simulated}
\subsubsection{Real data set}
\subsubsection{Novel data set}


\section{Results} \label{results:ca}
\subsection{Qualitative}

\subsection{Quantitative}
\subsubsection{Simulated data}
\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/sim_tiles}
\decoRule
\caption[Score of assemblies of simulated data sets]{Results of assemblies executed with simulated data sets.}
\label{fig:sim_tiles}
\end{figure}



\begin{table}[h!]
\caption{\textbf{Scores of assemblies of simulated data}}
\label{tab:scores_simulated}
\centering
\begin{tabular}{rlrrrrrrr}
  \hline
  & data set & CAP & CE & Fast-Plast & GetOrganelle & IOGA & NOVOPlasty & org.ASM \\ 
  \hline
  1 & sim\_150bp.0-1 & 79.10 & 100.00 & 99.48 & 100.00 &  & 91.52 & 100.00 \\ 
  2 & sim\_150bp.0-1.2M & 79.10 & 100.00 & 99.72 & 100.00 & 79.10 & 91.52 & 91.50 \\ 
  3 & sim\_150bp.1-10 &  & 56.44 & 100.00 & 76.98 &  & 91.52 & 78.00 \\ 
  4 & sim\_150bp.1-10.2M &  &  & 99.97 & 100.00 &  & 91.52 & 82.72 \\ 
  5 & sim\_150bp.1-100 & 75.72 & 100.00 & 99.48 & 100.00 & 66.09 & 91.52 & 91.50 \\ 
  6 & sim\_150bp.1-100.2M &  & 100.00 & 99.47 & 100.00 &  & 100.00 & 100.00 \\ 
  7 & sim\_150bp.1-1000 & 79.10 &  & 99.72 & 100.00 &  & 91.52 & 100.00 \\ 
  8 & sim\_150bp.1-1000.2M & 79.10 & 100.00 & 99.72 & 100.00 &  & 91.52 & 100.00 \\ 
  9 & sim\_250bp.0-1 & 79.10 & 100.00 & 93.82 & 100.00 &  & 91.52 & 91.50 \\ 
  10 & sim\_250bp.0-1.2M & 79.10 & 100.00 & 93.83 & 100.00 &  & 91.52 & 91.50 \\ 
  11 & sim\_250bp.1-10 &  & 54.98 & 68.45 & 78.89 & 52.71 & 91.52 & 40.20 \\ 
  12 & sim\_250bp.1-10.2M &  &  & 93.00 & 100.00 & 52.67 & 87.40 & 40.20 \\ 
  13 & sim\_250bp.1-100 & 72.81 & 100.00 & 93.82 & 100.00 &  & 87.40 & 100.00 \\ 
  14 & sim\_250bp.1-100.2M &  & 100.00 & 93.83 & 100.00 &  & 87.40 & 100.00 \\ 
  15 & sim\_250bp.1-1000 & 79.10 & 21.30 & 93.83 & 100.00 & 76.96 & 91.52 & 91.50 \\ 
  16 & sim\_250bp.1-1000.2M & 79.10 & 100.00 & 93.83 & 100.00 & 67.55 & 87.40 & 100.00 \\
  \hline
\end{tabular}
\end{table}




\subsubsection{Real data sets}
\begin{table}[h!]
\caption{\textbf{Mean scores of chloroplast genome assemblers}}
\label{tab:scores_real}
\centering
\begin{tabular}{rlrrrr}
  \hline
 & assembler & Median & IQR & N\_perfect & N\_tot \\ 
  \hline
1 & CAP & 45.25 & 50.19 &   0 & 369 \\ 
  2 & CE & 56.55 & 71.50 &  14 & 369 \\ 
  3 & Fast-Plast & 92.80 & 23.59 & 113 & 369 \\ 
  4 & GetOrganelle & 99.83 & 20.94 & 210 & 360 \\ 
  5 & IOGA & 71.10 & 11.21 &  0 & 338 \\ 
  6 & NOVOPlasty & 75.95 & 48.69 &  58 & 369 \\ 
  7 & org.ASM & 67.35 & 91.69 &  46 & 348 \\ 
   \hline
\end{tabular}
\end{table}





\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/swarm}
\decoRule
\caption[Scores of assemblies from real data sets]{Box and swarm plots depict the results from the scoring shown in \ref{eqn:score_ass}}
\label{fig:swarm}
\end{figure}




\subsubsection{Consistency}

\subsubsection{Real data sets}
\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/repro}
\decoRule
\caption[Comparison between two runs with the same assembler for consistency testing ]{Swarm plots depict the results from the scoring shown in \ref{eqn:score_ass} for two independent runs for each assembler on each of the data sets}
\label{fig:consisplot}
\end{figure}


\subsubsection{Novel}


\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/upset_novel}
\decoRule
\caption[Upset plot comparing the success rates for novel data sets]{nls}
\label{fig:upset_novel}
\end{figure}

\section{Discussion}

\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/upset}
\decoRule
\caption[Upset plot comparing the success rates of of all assemblers]{Upset plot showing the intersections of success rates between assemblers. A successful assembly was defined with a score > 99 according to equation \ref{eqn:score_ass}}
\label{fig:upset}
\end{figure}


