% Chapter 1
\newcommand{\formatprogramnames}[1]{\texttt{#1}}
\newcommand{\ce}{\formatprogramnames{chloroExtractor}}
\newcommand{\oa}{\formatprogramnames{ORG.Asm}}
\newcommand{\fp}{\formatprogramnames{Fast-Plast}}
\newcommand{\ioga}{\formatprogramnames{IOGA}}
\newcommand{\np}{\formatprogramnames{NOVOPlasty}}
\newcommand{\go}{\formatprogramnames{GetOrganelle}}
\newcommand{\cassp}{\formatprogramnames{Chloroplast assembly protocol}}



\chapter{Benchmarking of Chloroplast Genome Assembly Tools } % Main chapter title

\label{Chapter1} % For referencing the chapter elsewhere, use \ref{Chapter1}
This chapter is oriented on \cite{freudenthal2019landscape}, which as
has been published on the preprint server bioR$\chi$iv and submitted
for peer review. Only the parts from the publication which the
author majorly contributed to are included. If not cited otherwise the
plots were designed and generated by the author of this thesis.

%----------------------------------------------------------------------------------------

% Define some commands to keep the formatting separated from the content 
\newcommand{\keyword}[1]{\textbf{#1}}
\newcommand{\tabhead}[1]{\textbf{#1}}
\newcommand{\code}[1]{\texttt{#1}}
\newcommand{\file}[1]{\texttt{\bfseries#1}}
\newcommand{\option}[1]{\texttt{\itshape#1}}

%----------------------------------------------------------------------------------------

\section{Introduction} \label{intro:cp}
\subsection{Motivation}

Some organelles like mitochondria and chloroplasts contain their own genetic information
from which they are able to synthesize certain proteins independent of the nucleus
genome. Evolutionary this developed during endosymbiosis, a process which underlying
theory seeks to explain how eukaryotic cells formed from prokaryotes
\cite{mereschkowsky1905uber}; \cite{kutschera2005endosymbiosis}. This widely acknowledged
hypothesis explains how in the early evolution of eukaryotes other cells were
incorporated, which ultimately became organelles. The most likely precursors of today's
chloroplasts were photosynthetic bacteria or similar organisms
\cite{archibald2015endosymbiosis}. This process left its traces in the structure of
chloroplast genomes until today, which resemble more closely prokaryotic genomes than that
of its eukaryotic host cells. A typical chloroplast genome consists of circular DNA with a
size between 120 kbp to 160 kbp \cite{palmer_1985}, while plant core genomes are linear,
organized in chromosomes and larger by multiple orders of magnitude. \\
The first chloroplasts have been sequenced as early as 1986 and were isolated from
\textit{Marchantia polymorpha} and \textit{Nicotiana tabacum}
\cite{ohyama_chloroplast_1986}; \cite{shinozaki_complete_1986}. Complete reviews on the
structure of chloroplast genomes were authored by \cite{green_chloroplast_2011} and
\cite{wicke_evolution_2011}. Chloroplast genomics is widely applied in evolutionary
studies aiding to elucidate the processes involved in endosymbiosis and the development of
photosynthetic plants \cite{martin_evolutionary_2002};
\cite{xiao-ming_inferring_2017}. Over the course of natural adaptation, the plastid genome has
been reduced in size through endosymbiotic gene transfer, a form of horizontal gene
transfer, where fractions of plastid genomes are incorporated in the core genome
\cite{martin_evolutionary_2002}; \cite{deiner_environmental_2017}. This mechanism of
evolution is still ongoing an can be observed \textit{in vitro} \cite{bock2017witnessing};
\cite{fuentes2014horizontal}; \cite{stegemann2009exchange}. \\
In the case of \textit{Arabidopsis thaliana}, this process resulted in 14 \% of the core
genome's genes previously being located on the chloroplast, while 100-120 genes remain on
the chloroplast itself \cite{wicke_evolution_2011}, which by far would not suffice to
allow the chloroplast to function independently of its host cell. The fact that organelle
genomes are much smaller and highly conserved with a large gene content leads to
polymorphisms being more likely to cause functional changes in physiological
processes. Another difference between organelle and core genomes is that single
chloroplasts contain up to hundreds of copies of its own genome \cite{kumar_2014};
\cite{bendich_1987}. Considering that photosynthetic active cells again contain multiple
chloroplasts means that the number of chloroplast genomes therefore is considerably higher than the number of core genomes per cell. \\
Structurally, chloroplast genomes are made up of four distinct regions. Two inverted
repeats (IR), $IR_A$ and $IR_B$, ranging from 10 kbp to 76 kbp in size that divide the
circular genome into two regions: the large
single copy (LSC) and the small single copy (SSC) as shown in figure \ref{fig:cpast_genome} \cite{palmer_1985}.\\
Taking into account that the majority of assembly tools has been designed to assemble
linear core genomes, the structure of chloroplast genomes is an obstacle for many assembly
pipelines to overcome. This holds true especially for solving and aligning the IRs
correctly \cite{Wang2018}.

\begin{figure}[H]
\centering
\includegraphics[height=.55\textheight, width=.95\textwidth]{Figures/cpast}
\decoRule
\caption[Structure of a chloroplast genome]{Structure of the chloroplast genome of
  \textit{A. thaliana} with small single copy region (SSCR) and large single copy region
  (LSCR).The number denotes the length of the genome and its parts in kilo base pairs
  (kbp). Graphic from \cite{olejniczak2016chloroplasts}}
\label{fig:cpast_genome}
\end{figure}

Another difficulty for the assembly is heteroplasmy, which describes the phenomenon of
co-existence of multiple versions of the chloroplast's genome in a single organism and
even single cells of that respective organism. Heteroplasmy complicates genome assemblies
and ongoing from there the downstream analyses \cite{corriveau_1988}; \cite{Chat2002}. The
underlying evolutionary mechanisms behind heteroplasmy are not fully elucidated and
existing fitness advantages fueling heteroplasmy cannot be explained satisfactory by
standard evolutionary methods
\cite{Scar2016}. \\

Derived from a multitude of plant genome projects, there is a large variety of databases,
containing short read data for species without assembled organelle genomes available,
e.g. NCBI's sequence read archive (SRA) \cite{SRA2010}. Because most plant DNA extraction
protocols applied to procure raw input for sequencing use green leaf tissue as their
basis, they also contain a large amount of plastid DNA, providing a valuable source for
organelle genome assembly pipelines. In the course of this chapter, the performance of
these pipelines will be assessed.\\
Having larger numbers of assembled and annotated chloroplast genomes publicly available
will be beneficial for evolutionary studies and are a useful addition to bar-coding and
super-barcoding \cite{coissac_barcodes_2016}, aside from other biotechnological
applications \cite{daniell_chloroplast_2016}. To obtain those there is a variety of tools
available. In the course of this chapter the availability, usability and overall
performance of seven of those assembly pipelines will be assessed. Ultimately, the newly
gained insights will be utilized to attempt to assemble more than 100
chloroplasts \textit{de novo}.

\subsection{Extraction of chloroplast reads from whole genome data and general assembly workflow}
There is large array of strategies to assemble chloroplast genomes from raw sequencing
data \cite{twyford_strategies_2017}. In general the process of chloroplast genome assembly
involves three steps:
\begin{enumerate}[(i)]
\item extraction of plastid reads from the whole-genome sequencing (WGS) data
\item assembly of the plastid genome
\item solving the circular structure of
the genome with the IRs.
\end{enumerate}
There are three approaches to address step (i): the first one is to map all reads to a
reference chloroplast \cite{Vinga2012}, which works reasonably well if there is one
available for the same or at least a closely related species. The second approach is to
make use of the much larger coverage of chloroplast DNA compared to core DNA with a k-mer
analysis \cite{Chan2013}, this is for example done by \ce \hspace{0.25ex}, one of the tools used in this study
\cite{j_ankenbrand_chloroextractor:_2018}. The third way to accomplish plastid DNA
extraction is to combine both methods as done by \np \hspace{0.25ex}
\cite{dierckxsens_novoplasty:_2017}. Figure \ref{fig:cpast_workflow} shows the general
workflow of chloroplast assembly tools with the bifurcation at step (ii).

\begin{figure}[H]
\centering
\includegraphics[height=.65\textheight, width=.95\textwidth]{Figures/CE_workflow}
\decoRule
\caption[Chloroplast genome assembly workflow]{Standard workflow of chloroplast genome assembly. Graphic from \cite{j_ankenbrand_chloroextractor:_2018} }
\label{fig:cpast_workflow}
\end{figure}

\subsubsection{Purpose and scope of benchmarking the landscape of chloroplast assembly tools}

The purpose of this study is to provide insights into the landscape of chloroplast
assembly tools, to recommend best practices for organelle genome assemblies and to
contribute \textit{de novo} assemblies for many species and families without a reference
chloroplast available so far to the scientific community. \\



%----------------------------------------------------------------------------------------
\section{Material and methods}
\subsection{Methods}
\subsubsection{Data and code availability}
All the source code and data used is publicly available under the terms of the MIT-License. The source code
has been published on github \cite{github-benchmark-repo} and archived on zenodo \cite{zenodorepo} . The
docker images are available on dockerhub \cite{dockerhub-benchmark}.

\subsubsection{Tools}
To be included into this study, the software, including the source code, must be publicly
available. The study was further restricted to paired-end Illumina data sets as their sole
input source because they were abundantly available for this benchmark.  The only
technical requirement was being able to assemble chloroplast genomes from those paired-end
Illumina reads. The other requirements were dictated by reproducibility The software must
be open-source and available under the terms of a liberal software license and the
software must be able to be operated from a command line, since GUI-only tools are not
suited for highly repetitive, automated analyses. In total there were seven tools that met
those conditions:

\begin{enumerate}[(i)]
\item \ce \hspace{0.25ex} \cite{j_ankenbrand_chloroextractor:_2018}
\item \cassp \hspace{0.25ex} \cite{sancho_comparative_2018}
\item \go \hspace{0.25ex} \cite{jin_getorganelle:_2018}
\item \oa \hspace{0.25ex} \cite{coissac_barcodes_2016}
\item \ioga \hspace{0.25ex} \cite{bakker_herbarium_2016}
\item \fp \hspace{0.25ex} \cite{mckain__fast-plast_2017}
\item \np \hspace{0.25ex} \cite{dierckxsens_novoplasty:_2017}
\end{enumerate}


\subsubsection{Standardization and reproducibility}

Along with the study, easy and ready-to-use versions of all the involved programs, working
with standardized input, were published. For this purpose \texttt{docker} containers
\cite{merkel2014docker} were implemented. To work with the containers in a closed HPC
environment they were transformed into related \texttt{singularity} containers
\cite{kurtzer2017singularity}. To apply the programs users simply need to provide two files:
one for the forward reads (\texttt{forward.fq}) and one for the reverse reads
(\texttt{reverse.fq}) and run the containers without any need for further configuration or
installation besides \texttt{docker} or \texttt{singularity} itself, which can be easily
done on all popular operating systems. Both files are required to be in FASTQ
format. Besides the individual output files recording the process of the respective
program, all programs write the assembly products into files called \texttt{output.fa} in
FASTA format. For the quantitative and consistency measurements the singularity containers
were run on the Julia HPC-cluster of the University of W\"{u}rzburg using the SLURM
workload manager \cite{Jette02slurm}. All runs for all assemblies were set with a time
limit of 48 hours. This was necessary because some assemblers e.g. \ioga, if they did not finish
after at least 12 hours, showed the tendency not to finish, even after weeks of running.

\subsection{Data}
Three different data sets were used:
\begin{enumerate}[(i)]
\item simulated data from \textit{A. thaliana} chloroplasts 
\item real data with known reference chloroplast to rate the success of the assemblies 
\item novel data sets from NCBI's SRA without a known reference chloroplast to apply the
  gained knowledge to the \textit{de novo} assembly of more than 100 chloroplasts.
\end{enumerate}
\subsubsection{Simulated data}

To allow full control over all the parameters involved, we started with the simulated data.
In the present case the data's input parameters, thought to be influential on the outcome,
were: the read length, the ratio between chloroplast and core genome reads as well as the
total size of the data set. The data simulations were based on real data from the TAIR10
genome of \textit{A. thaliana} \cite{tair10} and spawned using \texttt{seqkit}
\cite{seqkit}. Core to chloroplast ratios simulated were: 0:1, 1:10, 1:1000 and 1:1000,
with read length of 150 and 250 bp. The artificial data consisted either of 2 million read
pairs or the full data available. The simulation process was documented and the code and
the data is available on github and zenodo \cite{zenododataset}.

\subsubsection{Real data set}\label{sec:cp_real}

Real data was selected from the SRA database. Table \ref{tab:sra_real} lists the search
parameters that had to be met for a plant to be included in the study from SRA.

\onehalfspacing
\begin{table}[H]
\caption{Data selection criteria for real data sets from SRA}
\label{tab:sra_real}
\centering
\begin{tabular}{lll}
  \toprule
  Choice & Option & Explanation \\
  \midrule
   Organism   & green plants & include only photosynthetic plants e.g. no algae  \\
   Strategy   & wgs          & only data from wgs projects included \\
   Platform   & Illumina     & include only paired-en Illumina reads \\
   Properties & biomol DNA   & include only biomol. DNA samples (e.g. no RNA) \\
   Layout     & paired       & exclude single-end reads  \\
   Selection  & random       & \\
   Access     & public       & only publicly available data included \\
  \bottomrule                                       
\end{tabular}
\end{table}
\doublespacing
\noindent
In total this resulted in 369 data sets representing a broad variety of the plant kingdom with many different
families and genera included.

\subsubsection{Novel data sets}

To assess the performance of assemblies without a published chloroplast on \texttt{CpBase}
\cite{cpbase} 105 data sets were selected from SRA. It was emphasized that the chosen read
libraries were as distant as possible related to the next relatives with a reference
chloroplast, related as possible in taxonomic terms according to NCBI
\cite{ncbitaxonomy}. This was achieved by a phlyogenetic analysis of the accessible data
sets on SRA by Frank F\"orster described in \cite{freudenthal2019landscape}.

\subsection{Evaluation}
\subsubsection{Quantitative}
Where applicable, each assembly from each assembler was compared to their respective
reference genome by alignment using \texttt{minimap2} \cite{li2018minimap2}. Based on
those alignments scores were calculated following equation \ref{eqn:score_ass} from 0 to
100, with 100 being a perfect score.

\begin{equation}
  score = \frac{1}{4} \cdot \left( cov_{ref} +  cov_{qry} + min\left\{ \frac{cov_{qry}}{cov_{ref}}, \frac{cov_{ref}}{cov_{qry}}\right\} + \frac{1}{n_{contigs} }\right) \cdot 100
  \label{eqn:score_ass}
\end{equation}

Four different metrics contributed equally to the final score:

\begin{enumerate}[(i)]
\item the coverage of the assembled genome compared to the reference genome $cov_{ref}$ as
  an estimate for the completeness
\item the vice versa case $cov_{qry}$ as a measure for the correctness of the assembly 
\item the success of resolving the IR correctly, estimated by the difference from the
  reference and the newly assembled genome
  $min\left\{ \frac{cov_{qry}}{cov_{ref}}, \frac{cov_{ref}}{cov_{qry}}\right\}$
\item the number of total contigs were weighted as $\frac{1}{n_{contigs}}$ giving a
  chloroplast with one contig the optimal score.
\end{enumerate}

While it is difficult to evaluate the success or failure of assemblies on a continuous
scale, equation \ref{eqn:score_ass} allows for objective and unbiased measurements. SNPs
or other small variants do not influence the outcome of the score because they are more
likely due to in-species variation between plastid genomes and not caused be the assembly
itself. Even if the latter is true it would be difficult to determine.
  
\subsubsection{Consistency}
For any given bioinformatical application consistency is a desired trait. Software ideally
should repeatedly yield the same output when provided with the same input and assembly
tools are exception. To evaluate the reproducibility of the seven tools, for all the 369 real
data sets described in section \ref{sec:cp_real}, they were assembled and scored twice
with each assembler. The correlations between the first and the second run's scores were
used as the measure for the robustness of a program.

\section{Results} \label{results:ca}
\subsection{Quantitative}
\subsubsection{Simulated data}
\label{results:sim}

The simulated data sets were assembled and scored with all the tools as described
above. Figure \ref{fig:sim_tiles} shows a tile plot with the results displaying a color
scale from orange over light
green to dark green representing the scores from 0 to 100. Blank spaces indicate the failure to produce any output in the given time frame of 48 hours. \\
While at first sight there is no clear correlation between the input data sets and the
score, it is clearly visibile that there are grave differences between the assemblers. Two
programs, namely \cassp \hspace{0.25ex} and \ioga, failed to correctly assemble a single
chloroplast's genome. \ioga \hspace{0.25ex} even fails to provide an output at all for the
majority of the data sets. While those two stand out as negative examples, \fp
\hspace{0.25ex} and \go \hspace{0.25ex} stand out as positive examples, perfectly or nearly
perfectly assembling all the data sets, with \go \hspace{0.25ex} surpassing the performance
of \fp. In the middle of the filed are \ce, \oa \hspace{0.25ex} and \np \hspace{0.25ex}
performing reasonably well, but sometimes lacking to solve the IRs and the circular
structure.

\begin{figure}[H]
\centering
\includegraphics[height=.55\textheight, width=.99\textwidth]{Figures/sim_tiles}
\decoRule
\caption[Score of assemblies of simulated data sets]{Results of assemblies executed with
  simulated data sets. The tile colors from orange to green indicate the score from 0 to
  99. Dark green tiles are score >99. Blank tiles point to assemblies, which failed to
  provide an output file. The axis on the left shows the ratio between nucleic and
  plastid DNA, the one on the right the size of the data sets. On the top the read length
  in base pairs (bp) is given and in the bottom the seven different assemblers}
\label{fig:sim_tiles}
\end{figure}

There is a significant difference between the performance of the assemblers in general.
The varying input parameters, however, do not have as grave an influence as the choice of
the assembler. While \fp \hspace{0.25ex} deals with the shorter reads of 150 bp much better
than with the longer reads of 250 bp, the scores of the other assemblers do not seem to be
influenced by the read length. There is no difference between the full and the subsampled
data sets. And while all assemblers appear to be more challenged by low chloroplast to
core genome ratios of 1:10, beyond a ratio of 1:100 it does not affect the quality of the
assemblies. Table \ref{tab:scores_simulated} shows all the individual results for all data
sets and assemblers. For the fields with no entry the respective assembler failed to
provide an output.

\begin{table}[ht]
  \caption[Scores of assemblies of simulated data]{Scores of assemblies of simulated data,
    with CAP = \cassp; CE = \ce; FP = \fp; GO = \go; NP = \np; oA = \oa ; length in base
    pairs (bp).}
\label{tab:scores_simulated}
\centering
%\resizebox{\textwidth}{!}{
  \begin{tabular}{lclrrrrrrr}
    \toprule
    Set  & Length & Ratio  & CAP   & CE     & FP     & GO     & \ioga  & NP     & oA     \\ 
    \midrule
    full & 150  & 0-1    & 79.10 & 100.00 & 99.48  & 100.00 &       & 91.52  & 100.00 \\ 
    2M   & 150  & 0-1    & 79.10 & 100.00 & 99.72  & 100.00 & 79.10 & 91.52  & 91.50  \\ 
    full & 150  & 1-10   &       & 56.44  & 100.00 & 76.98  &       & 91.52  & 78.00  \\ 
    2M   & 150  & 1-10   &       &        & 99.97  & 100.00 &       & 91.52  & 82.72  \\ 
    full & 150  & 1-100  & 75.72 & 100.00 & 99.48  & 100.00 & 66.09 & 91.52  & 91.50  \\ 
    2M   & 150  & 1-100  &       & 100.00 & 99.47  & 100.00 &       & 100.00 & 100.00 \\ 
    full & 150  & 1-1000 & 79.10 &        & 99.72  & 100.00 &       & 91.52  & 100.00 \\ 
    2M   & 150  & 1-1000 & 79.10 & 100.00 & 99.72  & 100.00 &       & 91.52  & 100.00 \\ 
    full & 250  & 0-1    & 79.10 & 100.00 & 93.82  & 100.00 &       & 91.52  & 91.50  \\ 
    2M   & 250  & 0-1    & 79.10 & 100.00 & 93.83  & 100.00 &       & 91.52  & 91.50  \\ 
    full & 250  & 1-10   &       & 54.98  & 68.45  & 78.89  & 52.71 & 91.52  & 40.20  \\ 
    2M   & 250  & 1-10   &       &        & 93.00  & 100.00 & 52.67 & 87.40  & 40.20  \\ 
    full & 250  & 1-100  & 72.81 & 100.00 & 93.82  & 100.00 &       & 87.40  & 100.00 \\ 
    2M   & 250  & 1-100  &       & 100.00 & 93.83  & 100.00 &       & 87.40  & 100.00 \\ 
    full & 250  & 1-1000 & 79.10 & 21.30  & 93.83  & 100.00 & 76.96 & 91.52  & 91.50  \\ 
    2M   & 250  & 1-1000 & 79.10 & 100.00 & 93.83  & 100.00 & 67.55 & 87.40  & 100.00 \\
    \bottomrule
  \end{tabular}
   
\end{table}

\subsubsection{Real data sets}

Table \ref{tab:scores_real} summarizes the results from the assemblies of 369 data sets
with the seven assemblers. Similar to the scores of the previous section there is a
significant difference between the tools. Likewise \go \hspace{0.25ex} is the most
successful assembler by a large margin with 210 of 369 chloroplast genomes perfectly
assembled. It completely fails to provide output for only 9 data sets, resulting in a
median score >99.  Contrary \cassp \hspace{0.25ex} and \ioga \hspace{0.25ex} both failed to completely
assemble a single genome. The performance of \fp \hspace{0.25ex} is reasonably well in comparison,
completing approximately half as many genomes as \go \hspace{0.25ex} and being the only
other tool whose average score surpasses 90. Similar to the trials with the simulated
data in chapter \ref{results:sim} \ce, \hspace{0.25ex} \np \hspace{0.25ex} and \oa
\hspace{0.4ex} are in the middle of the field.

\begin{table}[H]
  \caption{Median scores of chloroplast genome assemblers with inter-quartile range (IQR)
    and the number of perfect scores (n\_perfect) compared to the total number of assemblies
    (n\_tot) providing an output}
\label{tab:scores_real}
\centering
\begin{tabular}{llrrrr}
  \toprule
   Assembler & Median & IQR   & n\_perfect & n\_tot \\ 
  \midrule
   \cassp    & 45.25  & 50.19 & 0          & 369    \\ 
   \ce       & 56.55  & 71.50 & 14         & 369    \\ 
   \fp       & 92.80  & 23.59 & 113        & 369    \\ 
   \go       & 99.83  & 20.94 & 210        & 360    \\ 
   \ioga     & 71.10  & 11.21 & 0          & 338    \\ 
   \np       & 75.95  & 48.69 & 58         & 369    \\ 
   \oa       & 67.35  & 91.69 & 46         & 348    \\ 
  \bottomrule
\end{tabular}
\end{table}


Figure \ref{fig:swarm} emphasizes the large differences between the assemblers shown in
table \ref{tab:scores_real}. The swarm plots show distinct bands for some assemblers
e.g. \np \hspace{0.25ex} and \oa, suggesting that multiple assemblies fail to be solved
into a single contig genome at a certain point. As thoroughly discussed in section
\ref{dis_cp}, solely from the swarm plot, it is debatable if all the tools should be
recommended to be used for the purpose they were designed for.

\begin{figure}[H]
\centering
\includegraphics[height=.5\textheight, width=.99\textwidth]{Figures/swarm}
\decoRule
\caption[Scores of assemblies from real data sets]{Box and swarm plots depicting the
  results from scoring of the assemblies for the real data sets as caluculated by equation
  \ref{eqn:score_ass}}
\label{fig:swarm}
\end{figure}

\subsubsection{Consistency}

Consistency testing was done by re-running every assembly for the real data sets and
comparison of the two scores. \ce \hspace{0.25ex} was the only tool that was 100 \%
consistent over both runs.The consistency plot (figure \ref{fig:consisplot}) for \fp
\hspace{0.25ex} and \np \hspace{0.25ex} results in arrowhead shaped plots. With
differences between the first and second run appearing in assemblies with the highest
scores. All other assemblers appear to produce the same output in the two runs, except if
either run failed to complete the assembly at all. This is less pronounced for \cassp
\hspace{0.25ex} and \go\hspace{0.25ex} and is a grave issue for \oa \hspace{0.25ex} and
\ioga.

\begin{figure}[H]
\centering
\includegraphics[height=.49\textheight, width=.95\textwidth]{Figures/repro}
\decoRule
\caption[Comparison between two runs with the same assembler for consistency testing
]{Swarm plots depict the results from the scoring shown in \ref{eqn:score_ass} for two
  independent runs for each assembler on each of the data sets}
\label{fig:consisplot}
\end{figure}


\subsubsection{Novel assemblies}
 
The final assessment in the evaluation of the assemblers was to test them on novel data
sets without a published chloroplast. This step is important for two reasons: (i) it is
possible that certain tools perform well on known chloroplasts because they have knowledge
of their structure, which would lead to a lack of generalization on unknown genomes. (ii)
To apply and test the gained insights with the goal of providing
the scientific community with a larger variety of published chloroplast genomes.\\
As in previous evaluations the most successful assembler was \go, with 49 out of 105 novel data sets completely assembled. \\
Lacking a reference genome for alignment, the success had to be defined differently and
equation \ref{eqn:score_ass} was not suitable to evaluate the novel assemblies. Metrics
influencing the score of the novel assemblies were the number of contigs, solving the IRs
and the size of the SSC and LSC (figure \ref{fig:upset_novel}. This, known to the author, might be biased and not true
for all chloroplasts and assumes that all chloroplast genomes evolved according to the
general structure described in chapter \ref{intro:cp}. Figure \ref{fig:upset_novel}
compares the results of the assemblies with at least one successful assembly.

\begin{figure}[H]
\centering
\includegraphics[height=.45\textheight, width=.95\textwidth]{Figures/upset_novel}
\decoRule
\caption[Upset plot comparing the success rates for novel data
sets]{Upset plot comparing the success (single contig, length $\geq$
  130 kpb, ir $\geq$ 17 kbp) rates of the different assemblers for the
  novel data sets. The colored, horizontal bar plot show the total
  amount of successful assemblies for each assemblers. The black,
  vertical barplot the size of the intersection indicated by the dots
  in the middle}
\label{fig:upset_novel}
\end{figure}

\section{Discussion} \label{dis_cp}

The study presented in this chapter so far consists of two goals:
\begin{enumerate}[(i)]
\item to assess the overall performance of a variety of tools designed specifically for
  the assembly of circular chloroplast genomes from paired-end Illumina reads and
\item to \textit{de novo} assemble a variety of yet unpublished chloroplast genomes from existing data.
\end{enumerate}

To accomplish the first goal 16 simulated and 369 real data sets were used adding up to a
total of 5166 assemblies for the real data sets and 112 for the simulated data, along 735
assemblies for the novel data sets, thus underlying the statistical powers of this
benchmarking study.\\
The most successful tools were \go \hspace{0.25ex} and \fp, which are recommended to be
used complementary because, as shown in figure \ref{fig:upset}, they succeed for most data
sets compared to other assemblers and accomplish to satisfactory assemble chloroplast
genomes where the other fails. If both of them fail it might be worthwhile to repeat the
runs because other results could be expected as shown in the scatter plots of figure
\ref{fig:consisplot}, especially \fp \hspace{0.25ex} might be able to improve the
previously reached sore. Only if both of them fail it might be, even though improbable,
possible that \np \hspace{0.25ex} performs a success assembly. The other
assemblers should be used with caution. While \ce \hspace{0.25ex} might be good for a
quick overview due to its relatively low demand in computational time
\cite{freudenthal2019landscape}; \cassp, \oa \hspace{0.25ex} and \ioga \hspace{0.25ex} are
not recommended to be used as the primary tools in organelle genome assembly projects.

\begin{figure}[H]
\centering
\includegraphics[height=.65\textheight, width=.99\textwidth]{Figures/upset}
\decoRule
\caption[Upset plot comparing the success rates of of all assemblers]{Upset plot showing
  the intersections of success rates between assemblers. A successful assembly was defined
  by a score >99 according to equation \ref{eqn:score_ass}. The colored, horizontal
  barplot indicate the total number of successful assemblies for an individual tool. The
  black, vertical barplot gives the magnitude of the intersection between the assemblers
  indicated by the dots in the middle. Therefore the first and second vertical bars are to
  be interpreted as follows: 77 data sets were only successfully assembled by \go,
  likewise 48 genomes were assembled completely by \go \hspace{0.25ex} and \fp
  \hspace{0.25ex} and so on. }
\label{fig:upset}
\end{figure}


It might be possible that overall performance of a specific tool might change
significantly by fine tuning the input parameters of the tool, which was purposely not
done in the scope of the present study because this study was designed to mimic the
behavior of end-users and not developers of such tools. It was assumed that
users with little experience in bioinformatics are inclined to use the basic configurations of such a tool. \\

While there are huge differences between all assemblers, they are presented with the same
challenges and the bottlenecks are similar for all of them. However, the success rate of
passing those differs. Figure \ref{fig:alitv} shows the alignment of the genomes,
assembled with the seven tools, of \textit{Oryza brachyantha}, a grass distantly related
to cultivated rice \textit{Oryza sativa}, and the respective reference genome. For the
need of a linear representation of the circular genome the convention is to present
chloroplast genomes in the order LSC - IRa - SSC -IRb. \textit{O. brachyantha} was chosen
because multiple tools successfully or at least almost assembled the full genome.  \cassp
\hspace{0.25ex} is singled out, which only managed to assemble a few fragments on the SCC
and the IRs on many contigs. A common mistake is to return three contigs as \ioga
\hspace{0.25ex} did. They represent the LSC the SSC and one IR but failed to resolve those
regions into a one circular contig. \go \hspace{0.25ex} and \fp \hspace{0.25ex} were able
to reproduce the structure of the reference, while \ce \hspace{0.25ex} flipped the LSC and
\np \hspace{0.25ex} and \oa \hspace{0.25ex} were not able to construct the single contig
into the conventional structure. All of these are common mistakes appearing more or less
rare in all the assemblers. This could be a good starting point for the developers to
further improve their tools. In this example all but \cassp \hspace{0.25ex} were able to
construct all the parts of the chloroplast's genome, and the most common error was to
resolve the structure of the genome into a circular, one contig version.

\begin{figure}[H]
  \centering \includegraphics[height=.60\textheight, width=.99\textwidth]{Figures/AliTV.png} \decoRule
  \caption[AliTV plot of alignments of assemblies of \textit{Oryza brachyantha} of all
  assemblers]{ AliTV plot \cite{alitv} from \cite{freudenthal2019landscape} showing the
    alignments of \textit{Oryza brachyantha} chloroplast genomes for all seven
    assemblers. Regions in adjacent assemblies are connected with colored ribbons. The
    color codes of for the similarity between regions. The purple arrows indicate the IR
    regions}
\label{fig:alitv}
\end{figure}

\section{Conclusion \& outlook}

Organelle genomics is a promising field in plant genetics. As described in section
\ref{intro:cp} chloroplast genomes are well-suited for applications in evolutionary
sciences, taxonomy and barcoding applications. Alike for its mother branch core genomics,
for comparative chloroplast genomics it is just as crucial to obtain high quality
genomes. The quality is mainly influenced by two factors: the quality of the genome
sequencing protocol and the quality of the assembly. As shown the latter varies massively
between tools and not all tools are recommend equally from the conclusions drawn from the
experiments described above. All tools have room for improvement. This is not meant to
criticize the respectable work of the developers, but to encourage them to further develop
tools and publish them under terms of liberal software licenses for the greater benefit of
the entire scientific community.
%%% Local Variables:
%%% mode: latex
%%% TeX-master: "../main"
%%% End:
