\chapter{General Discussion and further Observations} % Main chapter title
\label{Chapter6}
\section{Genomic data preparation is error-prone}

Researching and applying quantitative genetics from genome assemblies to genomic selection
is tedious with many error-prone steps involved. To obtain optimal results every step in
the entire process has to be optimized individually, without loosing the larger frame out
of sight. \\
To perform analyses for quantitative genetics in general there are two types of data
required: (i) genotypic and (ii) phenotypic data. Both are equally important and take many
steps to procure. Figure \ref{fig:quan_flow} schematically displays the key steps involved
in obtaining genomic marker matrices for downstream analyses as GWAS and GS from selection
of candidate genotypes to the final numeric marker matrix. Genotyping can either be
achieved by whole genome sequencing or by SNP analysis with a SNP array.  The first step
after sequencing, which provides raw reads, is to assemble the genome. As discussed in
chapter \ref{Chapter1} genome assembly is a complicated process. This holds true for both
the assembly of core and plastid genomes. There is a large variety of tools available for
core genome assembly and like the ones for plastid genomes they vary in their algorithmic
approaches and likewise their accuracy \cite{zhang2011practical}, which makes it hard to
determine whether polymorphisms between individual genomes are due to artifacts in the
genome assembly pipeline or actually are mirrored in the biological genome. Furthermore,
genome assemblies result in one dimensional representations of formerly three dimensional
genomes, losing most of the spatial and epigenetic information.

\begin{figure}[H]
  \begin{center}
    \begin{tikzpicture}[node distance=2cm, scale=0.8, transform shape]
      \node (start0) [startstop] {Selection of suitable candidates};
      \node (start) [startstop,below of=start0] {DNA extraction};
      \draw [arrow] (start0) -- (start);
      \node (seq) [process, below of=start, xshift=-3cm] {Sequencing} ;
      \draw [arrow] (start) -- (seq);
      \node (SNP) [process, below of=start, xshift=3cm, yshift=-2cm] {SNP array} ;
      \draw [arrow] (start) -- (SNP);
      \node (ga) [process, below of=seq] {Genome assembly} ;
      \draw [arrow] (seq) -- (ga);
      \node (snpca) [process, below of=ga] {Alignment \& SNP calling} ;
      \draw [arrow] (ga) -- (snpca);
      \node (imp) [io, below of=snpca, xshift=3cm] {Imputation of missing values};
      \draw [arrow] (snpca) -- (imp) ;
      \draw [arrow] (SNP) -- (imp) ;
      \node (LD) [io, below of=imp] {LD pruning} ;
      \draw [arrow] (imp) -- (LD) ;
      \node (MAF) [io, below of=LD] {MAF filtering} ;
      \draw [arrow] (LD) -- (MAF) ;
      \node (bm) [startstop, below of=MAF] {Numeric marker matrix} ;
      \draw [arrow] (MAF) -- (bm) ;
    \end{tikzpicture}
    \caption[Schematic process of genotyping for quantitative genetics]{Schematic process of genotyping for quantitative genetics analyses with its crucial steps} \label{fig:quan_flow}
  \end{center}     
\end{figure}

After sequencing and assembling multiple genomes of a species the next step is to align
them to detect genetic polymorphisms such as SNPs, InDels, etc. followed by the imputation
of missing values. This takes into consideration that all the missing data are actually
missing due to the assembly and not actually missing in the genome as deletions. However,
this step is necessary because GWAS and genomic selection requires complete data without
missing values.  Again, there is a variety of tools for the imputation of missing
markers. In plant genomics the most commonly used software is Beagle
\cite{browning2007rapid}; \cite{browning2018one}, which is based on hidden Markov models.
As thoroughly reviewed by \cite{pook2019improving} the accuracy of the algorithm varies
vastly depending on the population, LD structure, chromosome region, effective population
size and the allele frequency, all possibly leading to errors adding up the ones already
introduced to the upper branches of the entire pipeline.

\subsection{Imputation leads to false positive GWAS results}

Faulty imputation and SNP calling can possibly result in false positive GWAS results as
shown in the following example. Data from phenotypic trials with 330 fully sequenced
\textit{A. thaliana} for carbon isotope discrimination were used to perform GWAS with a
marker matrix with 10 million SNPs imputed with Beagle 3.0
\cite{dittberner2018natural}. This resulted in one marker with a significant p value on
the fourth chromosome. Upon further investigation of the chromosomal region in question
using the unimputed data a complex haplotype structure was revealed as shown in figure
\ref{fig:chr_jul}.

\begin{figure}[H]
\centering
\includegraphics[height=.55\textheight, width=1.1\textwidth]{Figures/plot_NAs_AT}
\decoRule
\caption[Haplotype structure on a 1kb window of chromosome 4 of
\textit{A. thaliana}]{Haplotype structure on a 1 kb window of chromosome 5 of
  \textit{A. thaliana}. On the vertical axis the number of NAs in the population of 1135
  accessions for a given marker is displayed. The horizontal axis gives the physical
  position on the chromosome. Red markers are located in coding and blue markers in
  non-coding regions according to the TAIR10 annotation \cite{rhee2003arabidopsis}. The
  gray bars indicate more than five coherent missing values for one accession. The arrow
  points to the location of the significant GWAS hit.}
\label{fig:chr_jul}
\end{figure}

The significant SNP is located in a region where up to 80\% of the data were originally
missing values and were filled with Beagle 3.0. Additionally a complex structure of longer
or shorter deletions is present, completely cutting out the non-coding region between the
two coding ones. Taking a look at figure \ref{fig:chr_jul} it immediately becomes obvious
that imputation in this region has to be wrong because the complex haplotype structure is
a clear indication for the missing values not being due to sequencing errors, but that
they are actually mirrored in the biological genomes. The possibility of imputation
leading to false positives has been discussed by \cite{lin2010new}. The present case
provides an practical example of th phenomenon. \\
Further in the scope of the study it was assessed weather the phasing algorithm used in
Beagle 3.0 detected some signal from the haplotype structure that lead to the faulty
imputation. The different haplotypes and deletions were coded as pseudo-markers for
further association studies, all resulting in non-significant p-values. The plots in
figure \ref{fig:chr_jul} provide a good example to show how the information loss about
complex genomic structures can lead to false statistical assumptions.

\subsection{Numeric marker matrices cannot represent the complexity of genomes}

Figure \ref{fig:chr1} shows the complex haplotype structure of chromosome one of
\textit{A. thaliana}. The plots for chromosome two to five are included in appendix
\ref{haplo:str}. They basically all follow a similar pattern. The region directly flanking
the centromere is more polymorphic than the telomeres at the p and q arms of the
chromosomes, independent if the chromosome is metacentric like chromosome one and five,
telocentric as chromosome two and four acrocentric as chromosome three. The centromere
itself is highly conserved and generally coding regions have less haplotypes than
non-coding regions. E.g. on chromosome one over a 1 kbp window in 1135 accessions there are
ca. 78 different haplotypes in general and 98 in coding and 62 in non-coding regions one
average. The most polymorphic regions, however, are often coding regions, like a region on
the q arm of chromosome one located at around 22 Mbp, which has more than 700 segregating
haplotypes in the 1 kbp window. The region harbors a locus containing disease resistance
genes \cite{cheng2017araport11}, over the evolutionary advantages or disadvantages for
those regions being highly polymorphic can only be speculated.

\begin{figure}[H]
\centering 
\includegraphics[height=.55\textheight, width=1.1\textwidth]{Figures/chr1_hap}
\decoRule
\caption[Haplotype strutcture of chromosome 1 of \textit{A. thaliana}]{The number of
  segregating haplotypes with a polymorphism in at least one position over a stretch of 1
  kbp on chromosome one of \textit{A. thaliana}.}
\label{fig:chr1}
\end{figure}

Next to highly polymorphic regions there are regions, which are completely conserved and
do not have a single polymorphism in a 1 kpb window. Around the centromere there are
regions longer than 10 kpb with no SNPs. Intuitively one would assume that this would
indicate important household genes that do not allow for any alterations in the amino acid
sequence, however, the majority, around 75\% of those regions are considered to be
non-coding. Conserved non-coding sequences (CNS) have been widely studied and shown great
evolutionary importance and were witnessed across species with millions of years of
evolutionary distance \cite{Burgess946}. \\
The haplotype analysis allows to visualize another interesting evolutionary
artifact. Chromosome one of \textit{A. thaliana} was derived from a fusion of two
chromosomes of its next relative \textit{A. lyrata}. Next to the active centromere located
in the middle of the chromosome, at around 20 mbp there is a region that shares some
properties of a centromere, where one of the \textit{A. lyrata} centromeres was
located \cite{koch2007evolution}. \\
The haplotype as well as the LD structure overall or for special regions e.g flowering
time associated loci \cite{li2014multiple}, is two complex to be represented sufficiently
in a binary marker matrix. If his hold true, as shown, for \textit{A. thaliana} it matters
even more so in plants whose genomes are much larger, underwent multiple whole genome
duplications and consist of many more chromosome. Like the diploid \textit{Z. mays} with
10 chromosomes, the allotetraploid \textit{B. napus} a product of hybridization with 19
chromosomes from the two ancestral species \textit{B. rapa} and \textit{B. oleracea}
\cite{liu2018brassica}, or the even more complex allohexaploid genome of
\textit{T. aestivum} \cite{international2018shifting}.

\subsection{Input data for GWAS and GS}

Phenotypic trials can only ever represent a sub sample of whole populations. Even larger
trials in the 1001 genome project only feature a bit more than 1000 accessions
\cite{atwell2010}; \cite{1001genome}. In practical trials it is common to randomly pick
accessions, cultivars or genotypes in the hope that they will segregate for a certain
trait. Sometimes this can be backed by a PCA or an analysis of molecular variance to
choose suitable candidates \cite{holker2019european} but this is not common practice. This
results in allele frequencies in the subpopulations not following those of the global
populations and phenotypic values not following normal distributions. For the 402 tested
\textit{A. thaliana} traits analyzed only 72 follow a normal distribution (own
observation), according the Shapiro-Wilk test \cite{shapiro1965analysis}. Taking into
account that many statistical tests assume normal distributed data, this is an another
source of errors in the genomic analysis pipelines, leading to over or under inflation of
p-values. This effect can become very large for imbalanced and/or binary phenotypes like
YEL (appendix \ref{AppendixB}). One method to overcome this problematic is to use
permutation-based thresholds for significance, which can account better for phenotypic
distributions than Bonferroni thresholds (chapter \ref{Chapter3}). In the given example
the permutation threshold is around $10^{-16}$ and the Bonferroni threshold is
approximately $10^{-8 }$, potentially leading to a larger number of number of false
positive markers.\\
Due the many sources of statistical inaccuracies that can be possibly introduced in the
whole genome analysis pipeline all results have to be carefully evaluated, which often
times is not done sufficiently. For each significant marker that has been detected the raw
genomic information needs to assessed to validate the results. 


\section{Prospects in Genomic Selection and Plant Breeding and Conclusion}
Plant breeding, like in the last decades and centuries, will utilize the technology of its
time. Many new tools including genome editing were recently added to the tool box of
breeding and despite regulatory issues of quickly found its way to revolutionize modern
plant biotechnology \cite{araki2015towards}. While in the future GWAS and its relatives or
progeny will be used to further elucidate the nature of quantitative traits, genomic
selection and genome editing will allow further improvement of the worlds major crop
plants \cite{rodriguez2017engineering}, aiming to provide germplasms with the yield
potential required by the demand of the growing world's population.\\
Following the current trends in bioinformatics for plant breeding this full lead to a
further increase in the dimensionality of data, as genotyping costs will further decline
and modern automated phenotyping techniques allow for larger trials, swallowing less
resources. 


% Recombination and LD in \textit{A. thaliana} \cite{kim2007recombination}
% LD in \textit{A. thaliana} \cite{nordborg2002extent}
% Evolution of selfing \cite{tang2007evolution}
% Evolution and genetic differentiation among relatives of Arabidopsis thaliana \cite{koch2007evolution}
% FLC haplotypes 







%%% Local Variables:
%%% mode: latex
%%% TeX-master: "../main"
%%% End:
